docs/installation.rst

```diff
@@ -0,0 +1,143 @@
+Installation
+============
+
+Requirements
+------------
+
+- Python 3.10 or higher
+- IPython/Jupyter Notebook or JupyterLab
+- Operating System: Linux, macOS, or Windows
+
+Dependencies
+~~~~~~~~~~~~
+
+The following packages will be installed automatically:
+
+- ``ipython``: For magic command integration
+- ``psutil``: For I/O tracking on Linux and Windows
+- ``matplotlib``: For histogram visualization
+- ``numpy``: For data processing in histograms
+
+Installing from PyPI
+--------------------
+
+The easiest way to install iops-profiler is from PyPI using pip:
+
+.. code-block:: console
+
+   >> pip install iops-profiler
+
+Installing from Source
+----------------------
+
+To install the latest development version from GitHub:
+
+.. code-block:: console
+
+   >> git clone https://github.com/lincc-frameworks/iops-profiler.git
+   >> cd iops-profiler
+   >> pip install -e .
+
+Development Installation
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want to contribute to iops-profiler, install with development dependencies:
+
+.. code-block:: console
+
+   >> pip install -e '.[dev]'
+   >> pre-commit install
+
+This installs additional tools for testing, linting, and pre-commit hooks.
+
+Platform-Specific Setup
+-----------------------
+
+Linux
+~~~~~
+
+No additional setup required. The extension uses ``psutil`` which works out of the box.
+
+macOS
+~~~~~
+
```

Comment:
This section isn't strictly true. Every time the cell or line magic is run a new password prompt will appear. We should simply say this in the documentation without any justification.

IMO the entire "Important macOS notes" section can be stricken.